### ISSUE TITLE
Allow `property` usage in fake provider

### ISSUE THREAD

**Issue by voldemortensen** [external-secrets/external-secrets#1269]
**Describe the solution you'd like**

Currently the `fake` provider allows `key`, `value`, and `version` to be set in the provider.

```
spec:
  provider:
    fake:
      data:
        - key: "foo/bar"
          value: "baz"
          version: "v1"
```

I'm using the vault provider in my environments which supports requesting a `property` inside the vault path.

```
spec:
  data:
    - secretKey: foo_bar
      remoteRef:
        key: foo
        property: bar
```

I would like to be able to set up a fake provider for local development and access the properties so I don't have to have separate secrets for local dev.

**What is the added value?**

I could remove logic that uses different ExternalSecrets in local dev and production environments.

**Give us examples of the outcome**

```
# ClusterSecretStore
spec:
  provider:
    fake:
      data:
        - key: "foo"
           property: "bar"
           value: "baz"
---
# ExternalSecret
spec:
  data:
    - secretKey: foo_bar
       remoteRef:
         key: foo
         property: bar
---
# Secret
data:
  foo_bar: baz
```

**Observations (Constraints, Context, etc):**

N/A

**Comment by gusfcarvalho** [comment]
I believe a change to the CRD is not needed - rather we need to implement GJSON support here as well https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/fake/fake.go#L65-L72.

It should be easy to do, if anyone wants to tackle.

**Comment by shuheiktgw** [comment]
Hi, @gusfcarvalho 👋  I'm interested in contributing the eso project. Let me work on it as my first issue 🙂 
/assign

**Comment by shuheiktgw** [comment]
Created a PR https://github.com/external-secrets/external-secrets/pull/1285

**Comment by voldemortensen** [comment]
Why was this marked as completed?On Oct 20, 2022, at 20:13, github-actions[bot] ***@***.***> wrote:﻿
Closed #1269 as completed.
